src/components/Pagination/index.tsx

```diff
@@ -0,0 +1,269 @@
+import classNames from 'classnames';
+import { FC, PropsWithChildren } from 'react';
+export type PaginationProps = PropsWithChildren<{
+  className?: string;
+  currentPage: number;
+  totalPages: number;
+  onPageChange: (page: number) => void;
+  // Optional props
+  showIcon?: boolean;
+  displayFormat: 'pagination' | 'pagination-icon-only' | 'navigation' | 'navigation-group';
+}>;
+
+export const Pagination: FC<PaginationProps> = ({
+  currentPage,
+  totalPages,
+  onPageChange,
+  showIcon = false,
+  displayFormat = 'pagination',
+}) => {
+  if (displayFormat === 'pagination' || displayFormat === 'pagination-icon-only') {
```

Comment:
Here is just a suggestion. You don't need to agree or change it if you don't want... But, I'm not a big fan of `if-then-else` statement.

There is two possible way to improve this part of the code:

First option: Use switch-case
```ts
switch (displayFormat) {
    case 'pagination':
    case 'pagination-icon-only':
        return renderPagination(currentPage, totalPages, onPageChange, showIcon);
    case 'navigation':
        return renderNavigation(currentPage, totalPages, onPageChange, showIcon);
    default:
        return renderNavigationGroup(currentPage, totalPages, onPageChange, showIcon);
}
```

Second option: Just use if, 'cause you are returning, so you don't need else-if
```ts
if (displayFormat === 'pagination' || displayFormat === 'pagination-icon-only') {
    return renderPagination(currentPage, totalPages, onPageChange, showIcon, displayFormat);
} 
if (displayFormat === 'navigation') {
    return renderNavigation(currentPage, totalPages, onPageChange, showIcon);
}
return renderNavigationGroup(currentPage, totalPages, onPageChange, showIcon);
```

Reply:
I prefer the switch-case suggestion

Reply:
Thanks. I have refactor the code and used the switch case.